jooq/build.gradle.kts

```diff
@@ -37,11 +37,11 @@ gradlePlugin {
   }
 }
 
-java { targetCompatibility = JavaVersion.VERSION_19 }
+java { targetCompatibility = JavaVersion.VERSION_20 }
```

Comment:
Does this mean that we always support 1 version behind the current version?

Reply:
> Does this mean that we always support 1 version behind the current version?

Not always, but often. This is really the target compatibility of the Kotlin compiler version we're using, and the Kotlin complier tends to take a little time to add support for newly-released JVM versions. So when we upgrade Kotlin to a version that supports generating Java 21 bytecode, we can bump this up to 21.

It doesn't matter too much on a practical level since the JVM has rock-solid backward compatibility.